src/equations/kdv_1d.jl

```diff
@@ -0,0 +1,183 @@
+
+The semidiscretization looks the following:
+```math
+\begin{aligned}
+  \eta_t+\sqrt{g D} D_1\eta+ 1 / 2 \sqrt{g / D} \eta D_1 \eta +  1 / 2 \sqrt{g / D} D_1 \eta^2 +1 / 6 \sqrt{g D} D^2 D_3\eta &= 0.
+\end{aligned}
+```
+where ``D_1`` is a first-derivative operator, ``D_3`` a third-derivative operator, and ``D`` the still-water depth.
+
+It conserves
+- the total water mass (integral of ``\eta``) as a linear invariant
+and if upwind operators (``D_3 = D_{1,+} D_1 D_{1,-}``) or wide-stencil operators (``D_3 = D_1^3``) are used for the third derivative, it also conserves
+- the energy (integral of ``1/2\eta^2``)
+
+for periodic boundary conditions.
+
+- Diederik Korteweg and Gustav de Vries (1895). 
+  On the change of form of long waves advancing in a rectangular canal, and on a new type of long stationary waves
+  [DOI: 10.1080/14786449508620739](https://doi.org/10.1080/14786449508620739)
+- Abhijit Biswas, David I. Ketcheson, Hendrik Ranocha and Jochen Schütz (2025)
+  Traveling-Wave Solutions and Structure-Preserving Numerical Methods for a Hyperbolic Approximation of the Korteweg-de Vries Equation
+  [DOI: 10.1007/s10915-025-02898-x](https://doi.org/10.1007/s10915-025-02898-x)
+"""
+struct KdVEquation1D{RealT <: Real} <: AbstractKdVEquation{1, 1}
+    gravity::RealT # gravitational acceleration
+    D::RealT # still-water depth
+    eta0::RealT # constant still-water surface
+end
+
+function KdVEquation1D(; gravity, D = 1.0, eta0 = 0.0)
+    eta0 == 0.0 || @warn "The still-water surface needs to be 0 for the KdV equations"
+    KdVEquation1D(gravity, D, eta0)
+end
+
+"""
+    initial_condition_convergence_test(x, t, equations::KdVEquation1D, mesh)
+
+A traveling-wave solution used for convergence tests in a periodic domain, here for dimensional variables.
+"""
+function initial_condition_convergence_test(x, t, equations::KdVEquation1D, mesh)
+    g = gravity(equations)
+    D = equations.D
+    c = 1.5 * sqrt(g * D)
+    A = 2 * D * (c - sqrt(g * D)) / sqrt(g * D)
+    K = 1 / 2 * sqrt(3 * A / D^3)
+    x_t = mod(x - c * t - xmin(mesh), xmax(mesh) - xmin(mesh)) + xmin(mesh)
+    eta = A * sech(K * x_t)^2
+    return SVector(eta)
+end
+
+"""
+    initial_condition_manufactured(x, t, equations::KdVEquation1D, mesh)
+
+A smooth manufactured solution in combination with [`source_terms_manufactured`](@ref).
+"""
+function initial_condition_manufactured(x, t, equations::KdVEquation1D,
+                                        mesh)
+    eta = 1 + exp(-t / 2) * sinpi(2 * (x - t / 2))
+    return SVector(eta)
+end
+
+"""
+    source_terms_manufactured(q, x, t, equations::KdVEquation1D, mesh)
+
+A smooth manufactured solution in combination with [`initial_condition_manufactured`](@ref).
+
+How it was calculated, is described in:
+https://github.com/NumericalMathematics/DispersiveShallowWater.jl/pull/198/files#r2090805751
```

Comment:
```suggestion
https://github.com/NumericalMathematics/DispersiveShallowWater.jl/pull/198#discussion_r2090805751
```
The other link didn't lead me directly to the relevant comment.